src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/MessageProducerImplUnitTest.java

```diff
@@ -53,7 +54,7 @@ void testSuccessfulWriteToTopic()
             throws ServiceException, ExecutionException, InterruptedException {
         when(producer.sendAndReturnFuture(any())).thenReturn(mockedFuture);
 
-        messageProducerImpl.writeToTopic(getDummyChipsRestInterfacesSend());
+        messageProducerImpl.writeToTopic(getDummyChipsRestInterfacesSend(), anyString());
```

Comment:
you shouldn't use an argument matcher as an input into the method you are testing, they are for use in verification. You should pass a legitimate topicName in and test that it is passed into the producer as part of the kafkaMessage.

Reply:
comment above still applies. You've modified the method to take any extra string parameter as the topicName but you haven't performed any tests against it. anyString() is from the mockito package and should be used with mockito functions only.  

I can modify the code and the test still passes -
`@Override
    public void writeToTopic(ChipsRestInterfacesSend chipsMessage, String topicName) throws ServiceException {
        try {
            logger.info(String.format("Message %s: Writing message to topic: %s", chipsMessage.getMessageId(), topicName));
            byte[] serializedData = avroSerializer.serialize(chipsMessage);
            Message kafkaMessage = new Message();
            kafkaMessage.setValue(serializedData);
            kafkaMessage.setTopic("THIS WILL CAUSE APP TO FAIL");
            kafkaMessage.setTimestamp(Long.valueOf(chipsMessage.getCreatedAt()));
            Future<RecordMetadata> future = producer.sendAndReturnFuture(kafkaMessage);
            future.get();
        } catch (IOException | ExecutionException e) {
            throw new ServiceException(e.getMessage(), e);
        } catch (InterruptedException ie) {
            Thread.currentThread().interrupt();
            throw new ServiceException("Thread Interrupted when future was sent and returned", ie);
        }
    }`

With the changes you've made to the method, it should set the kafkaMessage topic name to the value passed in as a parameter. To test this you'll need to capture the kafkaMessage and assert that it's topicName matches the topicName passed into writeToTopic.

Reply:
see https://github.com/companieshouse/chips-rest-interfaces-consumer/blob/eb08352c00388669d15dbba6aa7cedf4740865b2/src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/consumer/MessageConsumerImplTest.java#L60
for a Captor example

Reply:
Added a captor